test/openmic_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/joke'
+require './lib/user'
+require './lib/openmic'
+require 'pry'
+
+class OpenMicTest < Minitest::Test
+  def setup
+    @open_mic = OpenMic.new({location: "Comedy Works", date: "11-20-18"})
+    @sal = User.new("Sal")
+    @ali = User.new("Ali")
+    @joke_1 = Joke.new(1, "Why did the strawberry cross the road?", "Because his mother was in a jam.")
+    @joke_2 = Joke.new(2, "How do you keep a lion from charging?", "Take away its credit cards.")
+
+
+  end
+
+  def test_existence
+    assert_instance_of OpenMic, @open_mic
+  end
+
+  def test_attributes
+    assert_equal "Comedy Works", @open_mic.location
+    assert_equal "11-20-18", @open_mic.date
+    assert_equal [], @open_mic.performers
+  end
+
+  def test_welcome_and_perfomers_methods
+    @open_mic.welcome(@sal)
+    @open_mic.welcome(@ali)
+    assert_equal [@sal, @ali], @open_mic.performers
+  end
+
+  def test_repeated_jokes?
+    @open_mic.welcome(@ali)
+    @ali.learn(@joke_1)
+    @ali.learn(@joke_2)
+    assert_equal false, @open_mic.repeated_jokes?
+    # @ali.tell(@sal, @joke_1)
```

Comment:
Here, we probably want ali or sal to `tell` the same joke - indicated that the joke was repeated during the open mic 🙂